src/garvis/capability_broker.py

```
@@ -0,0 +1,303 @@
+
+    def expire(self, *, session_id: str = "default") -> int:
+        rows = self.connection.execute(
+            "SELECT request_id FROM approvals WHERE session_id=? AND state=? AND expires_at<=?",
+            (session_id, ApprovalState.PENDING.value, _iso(_now())),
+        ).fetchall()
+        if not rows:
+            return 0
+        ids = [str(row["request_id"]) for row in rows]
+        marks = ",".join("?" for _ in ids)
+        self.connection.execute(
+            f"UPDATE approvals SET state=?,resolved_at=? WHERE request_id IN ({marks})",
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)